src/community/assortativity.jl

```diff
@@ -0,0 +1,52 @@
+"""
+    assortativity(g)
+
+Return the [assortativity coefficient](https://en.wikipedia.org/wiki/Assortativity)
+of graph `g`, defined as the Pearson correlation of excess degree between
+the end vertices of all the edges of the graph.
+
+The excess degree is equal to the degree of linked vertices minus one,
+i.e. discounting the edge that links the pair.
+In directed graphs, the paired values are the out-degree of source vertices
+and the in-degree of destination vertices.
+
+# Examples
+```jldoctest
+julia> using LightGraphs
+
+julia> assortativity(star_graph(4))
+-1.0
+```
+"""
+function assortativity(g::AbstractGraph)
+    nue  = ne(g)
+    sjk = sj = sk = sjs = sks = 0
+    for d in edges(g)
+        j = outdegree(g, src(d)) - 1
+        k = indegree(g, dst(d)) - 1
+        sjk += j*k
+        sj  += j
+        sk  += k
+        sjs += j^2
+        sks += k^2
+    end
+    return assortativity_coefficient(g, sjk, sj, sk, sjs, sks, nue)
+end
+
+"""
+assortativity coefficient for directed graphs: 
+see equation (21) in M. E. J. Newman: Mixing patterns in networks, Phys. Rev. E 67, 026126 (2003), 
+http://arxiv.org/abs/cond-mat/0209450
+"""
```

Comment:
Are these intended to be exposed as part of the API? If so, they will need properly-formatted docstrings. If not, it might be worth turning these into multi-line comments instead.